File: app/src/main/java/com/micrositba/lifemonitor/BluetoothDeviceDataAdapter.java
package com.micrositba.lifemonitor;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import java.util.List;

public class BluetoothDeviceDataAdapter extends RecyclerView.Adapter<BluetoothDeviceDataAdapter.DeviceViewHolder>{

    //List of bluetooth devices
    private List<BluetoothDeviceData> availableDevicesList;
    //Item Click Listener associated with adapter
    private ItemClickListener clickListener;
    //Address to highlight
    private String addressInUse;
    //Context
    private Context mContext;

    BluetoothDeviceDataAdapter(List<BluetoothDeviceData> availableDevicesList, String addressInUse, Context mContext) {
        this.availableDevicesList = availableDevicesList;
        this.addressInUse = addressInUse;
        this.mContext = mContext;
    }

    @Override
    @NonNull
    public DeviceViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_device, parent, false);
        return new DeviceViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(DeviceViewHolder holder, int position) {
        //Set device info
        holder.name.setText(availableDevicesList.get(position).getName());
        holder.address.setText(availableDevicesList.get(position).getAddress());
        holder.itemView.setTag(position);
        //If address matches address in use, change indicator
        if(mContext != null && availableDevicesList.get(position).getAddress().equals(addressInUse)) {
            View view = holder.itemView.findViewById(R.id.device_status_indicator);
            view.setBackgroundColor(mContext.getResources()
                    .getColor(R.color.colorAccent));
        }
    }

    @Override
    public int getItemCount() {
        return availableDevicesList.size();
    }

    void setClickListener(ItemClickListener itemClickListener) {
        this.clickListener = itemClickListener;
    }

    public class DeviceViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        private TextView name;
        private TextView address;
        DeviceViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.device_name);
            address = itemView.findViewById(R.id.device_address);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            if (clickListener != null) clickListener.onClick(v, getAdapterPosition());
        }
    }
}

File: app/src/main/java/com/micrositba/lifemonitor/BluetoothDeviceDataAdapter.kt
package com.micrositba.lifemonitor

import androidx.recyclerview.widget.RecyclerView
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat

class BluetoothDeviceDataAdapter(
        //List of bluetooth devices
        private val availableDevicesList: List<BluetoothDeviceData>,
        //Address to highlight
        private val addressInUse: String?,
        //Context
        private val mContext: Context?)
    : RecyclerView.Adapter<BluetoothDeviceDataAdapter.DeviceViewHolder>() {

    //Item Click Listener associated with adapter
    private var clickListener: ItemClickListener? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DeviceViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_device, parent, false)
        return DeviceViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: DeviceViewHolder, position: Int) {
        //Set device info
        holder.name.text = availableDevicesList[position].name
        holder.address.text = availableDevicesList[position].address
        holder.itemView.tag = position
        //If address matches address in use, change indicator
        if (mContext != null && availableDevicesList[position].address == addressInUse) {
            val view = holder.itemView.findViewById<View>(R.id.device_status_indicator)
            view.setBackgroundColor(ContextCompat.getColor(mContext, R.color.colorAccent))
        }
    }

    override fun getItemCount(): Int {
        return availableDevicesList.size
    }

    fun setClickListener(itemClickListener: ItemClickListener) {
        this.clickListener = itemClickListener
    }

    inner class DeviceViewHolder (itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val name: TextView = itemView.findViewById(R.id.device_name)
        val address: TextView = itemView.findViewById(R.id.device_address)

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            clickListener?.onClick(v, adapterPosition)
        }
    }
}
